src/models/line_item.ts

```diff
@@ -68,7 +68,10 @@ export default class LineItem extends Model<
   @Column
   public offerPalletId!: number
 
-  @BelongsTo(() => Pallet, 'offerPalletId')
+  @BelongsTo(() => Pallet, {
+    foreignKey: 'offerPalletId',
+    onDelete: 'CASCADE',
```

Comment:
@deammer this is all you need. It lets the database take care of deleting all entries that reference the pallet via a foreign key, when the pallet is deleted.

Reply:
That's amazing 🤯 thank you!